src/PrivateRoute.js

```diff
@@ -2,28 +2,26 @@ import React from 'react';
 import {
   Route,
   Redirect,
-  useLocation,
 } from 'react-router-dom';
 import PropTypes from 'prop-types';
 import { LOGIN_PATH } from './paths';
+import useAuth from './use-auth';
 
 function PrivateRoute(props) {
   const {
     path,
     exact,
     component,
-    loggedIn,
   } = props;
-  const location = useLocation();
+  const auth = useAuth();
 
   return (
-    loggedIn ? (
+    auth.user ? (
```

Comment:
There's apparently an ESLint rule that disallows casting to Boolean when using an object directly as part of a condition, because the value is always cast to a Boolean anyway. https://eslint.org/docs/rules/no-extra-boolean-cast
This code is considered fine:
```
  const loggedIn = !!auth.user;

  return (
    loggedIn ? (
```
But this isn't:
```
  return (
    !!auth.user ? (
```
Will use the first one.

Reply:
Ah ok, I didn't know that rule existed, your alternative looks much better because it's also easier to read. It might look obvious like this but if you have 3 or more conditions like that, code will be very hard to read without meaningful condition names, I would even go one step further and rename it `isLoggedIn` 

It's always best to name boolean variables in a name that sounds like asking a question that can only be answered with true or false, i.e. `isSomething`, `hasSomething`, `shouldSomething`,...

Reply:
Here are some guides and articles that could be useful in this topic:
https://github.com/airbnb/javascript#naming-conventions
https://www.codereadability.com/refactoring-if-statements-for-readability/